src/zero_copy/input.rs

```diff
@@ -222,7 +222,7 @@ impl<'a, 'parse, I: Input + ?Sized, E: Error<I>, S> InputRef<'a, 'parse, I, E, S
         }
     }
 
-    pub(crate) fn next(&mut self) -> (usize, Option<I::Token>) {
+    pub fn next(&mut self) -> (usize, Option<I::Token>) {
```

Comment:
Part of me wonders whether we should stabilise the offset being returned. Perhaps we could have a `next_token` input that's just a light wrapper around `next` but drops the first field? What do you think?

Reply:
Currently the `usize` is used for `Located::at`, however, if we changed it to `Marker` we could drop the first field. I like the idea, it further reduces the chance of issues with user-facing API lock-in

Reply:
Sure! That seems like a sensible solution.